python/build.py

```
@@ -574,8 +803,25 @@ def main():
+            print(f"  [checkpoint saved: {ckpt_path}]")
+
+            # Generate samples at checkpoint time
+            if tokenizer is not None and gpu_model is not None:
+                try:
+                    samples = generate_samples(gpu_model, cfg, tokenizer, step)
+                    for s in samples:
+                        preview = s["completion"][:80].replace("\n", " ")
+                        print(f"  [sample] {s['prompt'][:40]}... → {preview}...")
+                    if jsonl:
+                        jsonl.log(event="sample", step=step, samples=samples)
+                except Exception as e:
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)